### PR TITLE
This PR adds integration tests for the `autonomy mint` and `autonomy service` command groups

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR

- Introduces base test class for chain interaction tests
- Adds more detailed tests for `autonomy mint` module
- Adds tests for `autonomy service` module

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: autonomy/chain/utils.py [hunk 1]
```diff
@@ -44,25 +44,39 @@ def resolve_component_id(
...
+            metadata_uri = registry_contracts.agent_registry.get_token_uri(
+                ledger_api=ledger_api,
+                contract_address=contract_address,
+                token_id=token_id,
+            )
+        else:
+            metadata_uri = registry_contracts.component_registry.get_token_uri(
+                ledger_api=ledger_api,
+                contract_address=contract_address,
+                token_id=token_id,
```
Comment (angrybayblade) [inline autonomy/chain/utils.py:68]: Will try to refactor this in clean up PRs

File: autonomy/cli/helpers/chain.py [hunk 2]
```diff
@@ -277,7 +289,7 @@ def activate_service(
     except ServiceRegistrationFailed as e:
         raise click.ClickException(str(e)) from e
 
-    click.echo("Service activated successfully")
+    click.echo("Service activated succesfully")
```
Comment (angrybayblade) [inline autonomy/cli/helpers/chain.py:292]: Typo

File: packages/valory/contracts/service_registry/contract.py [hunk 3]
```diff
@@ -157,6 +157,21 @@ def get_service_owner(
             service_owner=checksum_service_owner,
         )
 
+    @classmethod
+    def get_token_uri(
```
Comment (angrybayblade) [inline packages/valory/contracts/service_registry/contract.py:161]: Resolve token URI

File: scripts/check_copyright.py [hunk 4]
```diff
@@ -340,6 +341,7 @@ def _file_filter(file: Path) -> bool:
                     and "build" not in file.parts
                 )
             )
+            and "dummy_packages" not in file_str
```
Comment (angrybayblade) [inline scripts/check_copyright.py:344]: skip dummy packages from the copyright checks

File: tests/data/dummy_packages/packages.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+    "dev": {
```
Comment (angrybayblade) [inline tests/data/dummy_packages/packages.json:2]: Dummy packages for testing

File: tests/test_autonomy/test_cli/test_mint/base.py [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    "-c",
+    str(COST_OF_BOND_FOR_AGENT),
+    "--threshold",
+    str(THRESHOLD),
+)
+
+
+@skip_docker_tests
+@pytest.mark.usefixtures("registries_scope_class")
+class BaseChainInteractionTest(BaseCliTest):
```
Comment (angrybayblade) [inline tests/test_autonomy/test_cli/test_mint/base.py:101]: Base class for testing on-chain interaction functionalities

File: tests/test_autonomy/test_cli/test_mint/test_mint_components.py [hunk 7]
```diff
@@ -60,36 +66,178 @@ def setup(self) -> None:
...
+        ),
+    )
+    def test_metadata_retrive_failure(
+        self, package_id: PackageId, parameters: Tuple[str, ...]
+    ) -> None:
+        """Test connection error."""
+
+        _ = registry_contracts.component_registry
+        _ = registry_contracts.agent_registry
+        _ = registry_contracts.service_registry
```
Comment (angrybayblade) [inline tests/test_autonomy/test_cli/test_mint/test_mint_components.py:179]: Loading contracts to enable patching

**Comment by angrybayblade** [comment]
Failures has nothing to do with the PR

**Review by solarw — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**